reya-rust-sdk/src/http_provider.rs

```diff
@@ -429,11 +451,16 @@ impl HttpProvider {
 
         let builder = proxy.triggerAutoExchange(inputs);
 
-        let transaction_result = builder.send().await?;
-
-        trace!("End Trigger Auto-exchange");
-
-        eyre::Ok(transaction_result.tx_hash().clone())
+        match builder.send().await {
+            Ok(transaction_result) => {
+                trace!("End Trigger Auto-exchange");
+                return eyre::Ok(transaction_result.tx_hash().clone());
+            }
+            Err(e) => {
+                handle_auto_exchange_error(e);
```

Comment:
- handle_auto_Exchange_error looks like a general rpc error handling fn, correct? Is that  case can we make it general usable?
- The errors are only logged, would be better to pass them on to the next layer with the Err(...). Add the error string into the format so we can log it e.g. in data dog logger

Reply:
I just renamed the fn into `handle_rpc_error`. Also sending the error string upstream if decoded. see [2d88303](https://github.com/Reya-Labs/reya-rust-sdks-monorepo/pull/24/commits/2d883034e2956bb6b0fcdd9eace3cbab6af9b46c)